src/nyx.rs

```diff
@@ -0,0 +1,251 @@
+ * (cf. https://github.com/rtk-rs/sp3/graphs/contributors)
+ * This framework is shipped under Mozilla Public V2 license.
+ *
+ * Documentation: https://github.com/rtk-rs/sp3
+ *
+ * The Nyx feature is released under AGPLv3
+ * Copyright (C) 2021-onward Christopher Rabotin <christopher.rabotin@gmail.com> et al. (cf. AUTHORS.md)
+ * Documentation: https://nyxspace.com/
+ */
+use log::error;
+use thiserror::Error;
+
+use crate::prelude::{Duration, Epoch, SP3Entry, SP3Key, SP3, SV};
+
+use anise::{
+    constants::{
+        celestial_objects::{MOON, SUN},
+        frames::{EARTH_J2000, MOON_J2000},
+    },
+    prelude::Almanac,
+};
+
+use std::sync::Arc;
+
+use hifitime::{TimeSeries, Unit};
+
+use nyx_space::{
+    dynamics::{
+        DynamicsError as NyxDynamicsError, OrbitalDynamics, SolarPressure, SpacecraftDynamics,
+    },
+    md::trajectory::Traj,
+    propagators::Propagator,
+    Spacecraft,
+};
+
+/// [SP3] prediction specific errors.
+#[derive(Debug, Error)]
+pub enum PredictionError {
+    #[error("undetermined initial state")]
+    UndeterminedInitialState,
+
+    #[error("dynamics error: {0}")]
+    NyxDynamics(#[from] NyxDynamicsError),
+}
+
+impl SP3 {
+    /// Obtain a [Spacecraft] model, at desired [Epoch] for desired [SV]
+    pub fn spacecraft_model(&self, epoch: Epoch, sv: SV) -> Option<Spacecraft> {
+        self.spacecraft_model_iter(epoch)
+            .filter_map(|(sv_i, sc_i)| if sv_i == sv { Some(sc_i) } else { None })
+            .reduce(|k, _| k)
+    }
+
+    /// Obtain a [Spacecraft] model at desired [Epoch], for each satellite
+    pub fn spacecraft_model_iter(
+        &self,
+        epoch: Epoch,
+    ) -> Box<dyn Iterator<Item = (SV, Spacecraft)> + '_> {
+        Box::new(self.satellites_orbit_iter(EARTH_J2000).filter_map(
+            move |(orbit_t, orbit_sv, orbit)| {
+                if orbit_t == epoch {
+                    let sc_model = Spacecraft::builder().orbit(orbit).build();
+                    Some((orbit_sv, sc_model))
+                } else {
+                    None
+                }
+            },
+        ))
+    }
+
+    /// Obtain a predicted [Traj]ectory for each satellite.
+    ///
+    /// ## Input
+    /// - almanac: [Almanac] definition
+    /// - initial_epoch: possible initial [Epoch].
+    /// When undefined, we simply use the latest state in time.
+    /// - prediction_duration: [Duration] of the predicted [Trajectory]
+    ///
+    /// ## Returns
+    /// - ([SV], [Trajectory]) for each satellite
+    /// - [PredictionError]
+    pub fn trajectory_predictions_iter(
+        &self,
+        almanac: Arc<Almanac>,
+        initial_epoch: Option<Epoch>,
+        prediction_duration: Duration,
+    ) -> Result<Box<dyn Iterator<Item = (SV, Traj<Spacecraft>)> + '_>, PredictionError> {
+        let orbital_model = OrbitalDynamics::point_masses(vec![MOON, SUN]);
+
+        let srp_model = SolarPressure::new(vec![EARTH_J2000, MOON_J2000], almanac.clone())?;
```

Comment:
```suggestion
        let srp_model = SolarPressure::new(vec![EARTH_J2000], almanac.clone())?;
```

We can most likely disable the shadow casted from the Moon here. It may happen once every few months, but not enough to significantly alter the propagation of the spacecraft.